examples/hikari_lightbulb_basic_queue/music_plugin.py

```diff
@@ -37,16 +37,13 @@ async def track_exception(self, lavalink: lavasnek_rs.Lavalink, event: lavasnek_
 async def _join(ctx: lightbulb.Context) -> Optional[hikari.Snowflake]:
     assert ctx.guild_id is not None
 
-    states = plugin.bot.cache.get_voice_states_view_for_guild(ctx.guild_id)
-    voice_state = [state async for state in states.iterator().filter(lambda i: i.user_id == ctx.author.id)]
-
-    if not voice_state:
+    if not (voice_state := ctx.bot.cache.get_voice_state(ctx.guild_id, ctx.author_id)):
         await ctx.respond("Connect to a voice channel first.")
         return None
 
-    channel_id = voice_state[0].channel_id
+    channel_id = voice_state.channel_id
 
-    assert ctx.guild_id is not None
+    assert channel_id is not None
```

Comment:
When can channel_be None?

Reply:
In the case where they are are leaving a channel, but in that case they wont have a voice state cached, it would return `None` instead, so you can just assert it here to make typehinters happy

ref: https://www.hikari-py.dev/hikari/voices.html#hikari.voices.VoiceState.channel_id